### PR TITLE
Remove packages with CG alerts from E2E tests in 5.11.x

### PR DESCRIPTION
**Pull request by nkolev92**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/2502

Regression? Last working version:

## Description

This is making progress towards fixing CG alerts for this branch. 

The cherry-picked commits are: 

https://github.com/NuGet/NuGet.Client/commit/3a1efb8c72cd9936ca5de1032888fded9931a103
https://github.com/NuGet/NuGet.Client/commit/1b7fc29707f417366f36fa799eff29dbec21f734
https://github.com/NuGet/NuGet.Client/commit/81ca6c54482d690309340e7fdd3bb55de777f547
https://github.com/NuGet/NuGet.Client/commit/3114c7cff0116ccd815c3e0902bea7130ce3fb8c

Note that this doesn't fix all the issues in the branches, but some. 
I think even if this doesn't fix all issues, it should still be merged, as maintaining these branches will be constant work.

## PR Checklist

- [x] PR has a meaningful title
- [x] PR has a linked issue.
- [x] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [x] Test exception - This changes tests.
  - **OR**
  - [ ] N/A 

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [x] N/A

### CODE REVIEW

File: test/EndToEnd/tests/ServicesTest.ps1 [hunk 1]
```diff
@@ -654,12 +654,12 @@ function Test-InstallPackageAPIInstalledLowerVersionPackage
     $p = New-ClassLibrary
 
     # Act
-    [API.Test.InternalAPITestHook]::InstallPackageApi("microsoft.owin","2.0.0")
-    [API.Test.InternalAPITestHook]::InstallPackageApi("microsoft.owin","3.0.0")
+    [API.Test.InternalAPITestHook]::InstallPackageApi("nuget.librarymodel","6.2.0")
+    [API.Test.InternalAPITestHook]::InstallPackageApi("nuget.librarymodel","6.3.0")
```
Comment (zivkan) [inline test/EndToEnd/tests/ServicesTest.ps1:658]: this seems to have caused a test failure, as the target project is targeting net45, and is incompatible with the package.

**Review by zivkan — DISMISSED** [review]

**Review by jeffkl — DISMISSED** [review]

File: test/EndToEnd/tests/ServicesTest.ps1 (hunk 1 above)
Comment (nkolev92) [reply test/EndToEnd/tests/ServicesTest.ps1:658]: Yep, that seems like a true failure. I'm looking into it.

Comment (nkolev92) [reply test/EndToEnd/tests/ServicesTest.ps1:658]: I started with https://github.com/NuGet/NuGet.Client/commit/61713994c334ca0acc39dccdf2e08295cd39b1b0#diff-26005e27a6f30d3a28445c7720ff5a1ec00181c252735e4935b02b2b2afb58b7. 

I'll ping for a rereview once the tests are all passing.

Comment (nkolev92) [reply test/EndToEnd/tests/ServicesTest.ps1:658]: Interesting that cause binding redirect tests to start failing such as BindingRedirectClassLibraryWithDifferentDependents, BindingRedirectClassLibraryWithDifferentDependents and BindingRedirectDuplicateReferences. 

All 3 are failing as in https://github.com/NuGet/Home/issues/12292.

Comment (nkolev92) [reply test/EndToEnd/tests/ServicesTest.ps1:658]: I'll split the 2 templates. This might give us a hint about the binding redirect failures.

Comment (nkolev92) [reply test/EndToEnd/tests/ServicesTest.ps1:658]: I'll split the 2 templates. This might give us a hint about the binding redirect failures.

Comment (nkolev92) [reply test/EndToEnd/tests/ServicesTest.ps1:658]: Create separate templates. We already have a precedent with a 46 template.

**Review by jeffkl — DISMISSED** [review]

File: test/EndToEnd/tests/ServicesTest.ps1 (hunk 1 above)
Comment (nkolev92) [reply test/EndToEnd/tests/ServicesTest.ps1:658]: passing now

**Review by martinrrm — APPROVED** [review]

**Pull request merged**